### ISSUE TITLE
Assertion error "assert not isinstance(result, Deferred)" when I add a new builder and reconfigure

### ISSUE THREAD

**Issue by evpo** [buildbot/buildbot#5027]
I added a new build and called `buildbot reconfig master`.

My configuration ran successfully. Then, immediately after that Buildbot reconfigures workers. Since I added a new Builder it probably tries to contact the corresponding Worker. This is when the assertion fails, I believe. Buildbot continues working and all other builds work fine. However, all build requests for the new build get stuck in the Pending Buildrequests queue forever. Even though the worker is connected and functions.

If I restart the worker in question, it is attached to the Builder and all build requests start to build.

Versions:
```
[["Python","3.7.3"],["Buildbot","2.3.1"],["Twisted","19.7.0"]]
```

The exception extract from the log:
```
2019-10-02 16:50:40+1000 [Broker,2,172.17.0.1] during reconfig:

Traceback (most recent call last):
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/twisted/internet/defer.py", line 654, in _runCallbacks
current.result = callback(current.result, *args, **kw)
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/twisted/internet/defer.py", line 1475, in gotResult
_inlineCallbacks(r, g, status)
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
result = result.throwExceptionIntoGenerator(g)
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/twisted/python/failure.py", line 512, in throwExceptionIntoGenerator
return g.throw(self.type, self.value, self.tb)
---  ---
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/buildbot/master.py", line 397, in doReconfig
yield self.reconfigServiceWithBuildbotConfig(new_config)
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
result = result.throwExceptionIntoGenerator(g)
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/twisted/python/failure.py", line 512, in throwExceptionIntoGenerator
return g.throw(self.type, self.value, self.tb)
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/buildbot/util/service.py", line 49, in reconfigServiceWithBuildbotConfig
yield svc.reconfigServiceWithBuildbotConfig(new_config)
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
result = result.throwExceptionIntoGenerator(g)
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/twisted/python/failure.py", line 512, in throwExceptionIntoGenerator
return g.throw(self.type, self.value, self.tb)
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/buildbot/process/measured_service.py", line 31, in reconfigServiceWithBuildbotConfig
yield super(MeasuredBuildbotServiceManager, self).reconfigServiceWithBuildbotConfig(new_config)
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
result = result.throwExceptionIntoGenerator(g)
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/twisted/python/failure.py", line 512, in throwExceptionIntoGenerator
return g.throw(self.type, self.value, self.tb)
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/buildbot/util/service.py", line 520, in reconfigServiceWithBuildbotConfig
yield svc.reconfigServiceWithSibling(config_sibling)
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
result = result.throwExceptionIntoGenerator(g)
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/twisted/python/failure.py", line 512, in throwExceptionIntoGenerator
return g.throw(self.type, self.value, self.tb)
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/buildbot/worker/base.py", line 331, in reconfigServiceWithSibling
yield self.updateWorker()
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
result = g.send(result)
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/buildbot/worker/base.py", line 538, in sendBuilderList
d1 = self.attachBuilder(b)
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/buildbot/worker/base.py", line 543, in attachBuilder
return builder.attached(self, self.worker_commands)
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/twisted/internet/defer.py", line 1613, in unwindGenerator
return _cancellableInlineCallbacks(gen)
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/twisted/internet/defer.py", line 1529, in _cancellableInlineCallbacks
_inlineCallbacks(None, g, status)
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/twisted/internet/defer.py", line 1421, in _inlineCallbacks
status.deferred.callback(getattr(e, "value", None))
File "/home/mk7/ve/bb-master/sandbox/lib64/python3.7/site-packages/twisted/internet/defer.py", line 459, in callback
assert not isinstance(result, Deferred)
builtins.AssertionError:
 
2019-10-02 16:50:40+1000 [Broker,2,172.17.0.1] WARNING: reconfig partially applied; master may malfunction

```

**Comment by evpo** [comment]
It maybe related to #4757

**Comment by p12tic** [comment]
Fixed in #5029.
